controllers/SiteController.php

```diff
@@ -56,7 +57,13 @@ public function actionView()
     {
         return $this->render('view', [
             'newsDataProvider' => new ActiveDataProvider([
-                'query' => News::find()->orderBy('id desc'),
+                'query' => News::find()
+                    ->with(['tags' => function ($query) {
+                        /** @var ActiveQuery $query */
+                        $query->with('news');
+                    }])
+                    // ->with('tags.news') // the same as above
+                    ->orderBy('id desc'),
```

Comment:
to reduce the overhead of instantiating the News AR instances for tags the following `asArray()` call can be added:

```php
                'query' => News::find()->with(['tags' => function($q) {
                    /** @var ActiveQuery $q */
                    $q->with(['news' => function($q) {
                        /** @var ActiveQuery $q */
                        $q->asArray();
                    }]);
                }])->orderBy('id desc'),
```